lib/mix/lib/mix/task.compiler.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Mix.Task.Compiler do
+  @moduledoc """
+  This module defines the behaviour for a Mix task that does compilation.
+
+  A Mix compiler task can be defined by simply using `Mix.Task.Compiler`
+  in a module starting with `Mix.Tasks.Compile.` and defining
+  the `run/1` function:
+
+      defmodule Mix.Tasks.Compile.MyLanguage do
+        use Mix.Task.Compiler
+
+        def run(_args) do
+          :ok
+        end
+      end
+
+  If the compiler uses manifest files to track stale sources, it should
+  define `manifests/0`, and if it writes any output to disk it should
+  also define `clean/0`.
+
+  ## Attributes
+
+  The following attributes are used the same way as in other Mix tasks:
+
+  * `@shortdoc`  - makes the task public with a short description that appears on `mix help`
+  * `@recursive` - runs the task recursively in umbrella projects
```

Comment:
We usually indent bullet lists with two spaces and leave a blank line after them before the closing `"""`.